core/trino-main/src/main/java/io/trino/json/PathEvaluationVisitor.java

```diff
@@ -648,6 +651,42 @@ protected List<Object> visitIrDatetimeMethod(IrDatetimeMethod node, PathEvaluati
         throw new UnsupportedOperationException("date method is not yet supported");
     }
 
+    @Override
+    protected List<Object> visitIrDescendantMemberAccessor(IrDescendantMemberAccessor node, PathEvaluationContext context)
+    {
+        List<Object> sequence = process(node.getBase(), context);
+
+        return sequence.stream()
+                .map(object -> descendants(object, node.getKey()))
+                .flatMap(Collection::stream)
+                .collect(toImmutableList());
+    }
+
+    private List<Object> descendants(Object object, String key)
```

Comment:
This is not going to be very efficient with all the intermediate lists that are being created. Rewrite it using an "accumulator" and a method that appends to it as it walks the tree recursively.